scripts/mbedtls_dev/bignum_core.py

```diff
@@ -72,29 +72,56 @@ def generate_function_tests(cls) -> Iterator[test_case.TestCase]:
             yield cls(a_value, b_value).create_test_case()
 
 
+class BignumCoreOperationArchSplit(BignumCoreOperation):
+    #pylint: disable=abstract-method
+    """Common features for bignum core operations where the result depends on
+    the limb size."""
+
+    def __init__(self, val_a: str, val_b: str, bits_in_limb: int) -> None:
+        super().__init__(val_a, val_b)
+        bound_val = max(self.int_a, self.int_b)
+        self.bits_in_limb = bits_in_limb
+        if self.bits_in_limb == 32:
+            self.dependencies = ["MBEDTLS_HAVE_INT32"]
+            byte_len = 4*bignum_common.limbs_mpi4(bound_val)
+            self.hex_digits = 2*byte_len
+            self.bound = bignum_common.bound_mpi4(bound_val)
```

Comment:
The code of the `.function` files looks in good shape to me now (I haven't done a detailed code review).

The python code seems to have acquired some duplication though. Why are there separate sets of functions for `_mpi4` and `_mpi8` instead of a single function taking the limb size as argument? I'd expect the code here to be something like
```
LIMB_SIZE_DEPENDENCY = {32: 'MBEDTLS_HAVE_INT32', 64: 'MBEDTLS_HAVE_INT64'}
bound = bound_mpi(bound_val, bits_in_limb)
self.dependencies.append(LIMB_SIZE_DEPENDENCY[bits_in_limb])
```

Reply:
>The code of the .function files looks in good shape to me now 

That's great, I am very happy that the general shape of the code in the `.function` files looks good.

> (I haven't done a detailed code review).

That is perfectly fine, you gave exactly the kind of high level feedback I was looking for. Thanks!

> Why are there separate sets of functions for `_mpi4` and `_mpi8`

That part of the code is still under review in the prerequisite PR (https://github.com/Mbed-TLS/mbedtls/pull/6399), I'll raise your suggestion there.

I like the idea about the limb size dependency lookup table and I'll update this PR before moving it out from draft state.

Reply:
We unified the bound functions in the dependency PR. I tried to implement the lookup table as well, but it didn't look as pretty in the end and didn't work the first try. I'll give it another try when reworking the PR, but wouldn't like to spend too much time on this.